packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -381,6 +385,10 @@ class BottomNavigationBar extends StatefulWidget {
   ///    orientation.
   final BottomNavigationBarLandscapeLayout? landscapeLayout;
 
+  /// This flag is controlling how [BottomNavigationBar] is going to use
+  /// the colors provided by the properties and the theme.
```

Comment:
We should have a more complete description of exactly what this flag does.

Reply:
Is this needed because it breaks existing tests if it is not true by default?

Reply:
Yes, it is a change that would break your golden test. You could refer to the closed PR for our conversation on that matter. The flag was suggested by @Hixie in order to avid that and still have the changes implemented. The goldens break as soon I separate the color Tweens. This was a suggestion from you in the previous closed PR. It is correct a suggestion.